trantor/net/TcpClient.cc

```diff
@@ -66,10 +69,16 @@ TcpClient::TcpClient(EventLoop *loop,
     (void)validateCert_;
     connector_->setNewConnectionCallback(
         std::bind(&TcpClient::newConnection, this, _1));
-    connector_->setErrorCallback([this]() {
-        if (connectionErrorCallback_)
+    // WORKAROUND: somehow we got use-after-free error
+    auto guard = std::shared_ptr<TcpClient>(this, [](TcpClient *) {});
```

Comment:
this creates a shared_ptr with no relation to the original control block and is illegal